2 - RTB/Documentazione esterna/Analisi dei Requisiti/Analisi dei Requisiti.typ

```diff
@@ -0,0 +1,341 @@
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+- é stato caricato un file per la configurazione dell'ambiente;
+- tale file è stato aperto correttamente dal programma;
+- il programma ha ricavato informazioni non valide dal file.
+
+$bold("Postcondizioni: ")$
+viene visualizzato l'errore relativo al caricamento di un file con informazioni incongruenti.
+
+$bold("Scenario: ")$
+L'utente ha caricato un file per la configurazione dell'ambiente contenente informazioni incongruenti.
+
+
+
+== UC-2 Configurazione ambiente 3d manuale
+$bold("Descrizione: ")$
+configurazione manuale del perimetro dell'ambiente di lavoro.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+è stato dato inizio alla procedura di configurazione manuale dell'ambiente di lavoro.
+
+$bold("Postcondizioni: ")$
+- la forma e il perimetro dell'ambiente di lavoro è stato configurato manualmente;
+- l'ambiente così generato ha rimosso eventuali elementi precedentemente configurati.
+
+$bold("Scenario: ")$
+l'utente inserisce i dati relativi alla configurazione.
+
+$bold("Estensioni: ")$
+UC-2.1 Visualizzazione errore sui dati.
+
+
+
+=== UC-2.1 Visualizzazione errore sui dati
+$bold("Descrizione: ")$
+i dati inseriti per la configurazione manuale dell'ambiente di lavoro non sono validi.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+- inseriti dati per la configurazione manuale dell'ambiente;
+- tali dati non sono utilizzabili dal programma.
+
+
+$bold("Postcondizioni: ")$
+viene visualizzato l'errore relativo all'inserimento di dati non validi.
+
+$bold("Scenario: ")$
+l'utente inserisce dati relativi alla configurazione dell'ambiente non validi.
+
+
+
+== UC-3 Modifica ambiente 3d
+$bold("Descrizione: ")$ 
+il perimetro dell'ambiente di lavoro viene modificato successivamente alla sua configurazione iniziale.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+- almeno una configurazione dell'ambiente deve essere avvenuta con successo;
+
+$bold("Postcondizioni: ")$
+l'ambiente di lavoro è stato correttamente modificato in funzione delle richieste dell'utente.
+
+$bold("Scenario: ")$
+- l'utente avvia la modifica dell'ambiente di lavoro;
+- l'utente immette i dati richiesti.
+
+$bold("Estensioni: ")$
+UC-3.1 Visualizzazione errore di modifica dell'ambiente.
+
+
+
+
+=== UC-3.1 Visualizzazione errore di modifica dell'ambiente
+$bold("Descrizione: ")$ 
+i dati inseriti per la modifica dell'ambiente di lavoro non sono validi con quanto configurato precedentemente.
+
+$bold("Attore: ")$
+utente.
+
+$bold("Precondizioni: ")$
+- l'utente ha immesso i dati per la modifica dell'ambiente;
+- tali dati non sono congrui con la precedente configurazione dell'ambiente.
+
+$bold("Postcondizioni: ")$
+all'utente viene notificato l'errore relativo ad un'immissione errata dei dati per la modifica dell'ambiente.
+
+$bold("Scenario: ")$
+l'utente ha immesso dei dati errati per la modifica dell'ambiente.
+
+
+
+
+== UC-4 Creazione scaffale
```

Comment:
In questo use case, per quanto si parli della creazione dello scaffale, non si menzionano minimamente i bin associati, che devono anch'essi essere configurati (per dimensione e numero). Questa mancanza crea ambiguità con lo use case riguardo alla cancellazione dello scaffale, il quale menziona la necessità dei "bin vuoti" nonostante questi non siano mai stati menzionati prima